### PR TITLE
[cmp.alg] Clarify ill-formed CPO invocations are SFINAE-friendly

### LINKED ISSUE #4492: [cmp.alg] Notes for ill-formed cases are inconsistent with those in [range.access]
It's noted that "This case can result in substitution failure when `*_­order(E, F)` appears in the immediate context of a template instantiation." in [[cmp.alg]](https://eel.is/c++draft/cmp.alg), which implies that only one of the two ill-formed cases results in substitution failure. And such notes don't exist in specifications of `compare_*_order_fallback` CPOs.

Are the cases for hard error overriding [[customization.point.object]/4](https://eel.is/c++draft/customization.point.object#4) are intended?

If not, I think that we should also use the form "(Diagnosable) ill-formed cases above result in substitution failure when `*(E, F)` appears in the immediate context of a template instantiation" (from [[range.access]](https://eel.is/c++draft/range.access)) in [cmp.alg] and apply the notes for all six CPOs.

By the way, the uses of `std::` are also inconsistent in these notes currently.

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #4492

### CODE REVIEW

**Comment by tkoeppe** [comment]
Looks good, but I'll wait for approval from @jwakely and @CaseyCarter. Thank you!

**Review by jwakely — APPROVED** [review]

**Review by CaseyCarter — APPROVED** [review]
This looks great! +1 to removing the unconventional `std::` qualifications, and I agree that it's better to strike "Diagnosable" from the notes for these CPOs with no IFNDR cases than to keep the notes consistent with the notes for the range access CPOs.

**Comment by jensmaurer** [comment]
@tkoeppe , approvals have arrived.

**Pull request merged**
